tests/unit/test_kernel.py

```diff
@@ -19,1993 +19,572 @@
-        enable JIT decoration of methods inside this class.
-        """
-        return cls(*children, **aux_data)
-
-    def compute_elementwise(self, x: ArrayLike, y: ArrayLike) -> Array:
-        r"""
-        Evaluate kernel on two inputs ``x`` and ``y``.
-
-        We assume ``x`` and ``y`` are two vectors of the same dimension.
-
-        :param x: Vector :math:`\mathbf{x} \in \mathbb{R}^d`
-        :param y: Vector :math:`\mathbf{y} \in \mathbb{R}^d`
-        :return: Kernel evaluated at (``x``, ``y``)
-        """
-        return self.a * (x + y)
-
-
-class KernelNoTreeUnflatten:
-    """
-    Example kernel with no method to unflatten a pytree.
-
-    This kernel is used to verify handling of invalid inputs to Stein kernels.
-    """
-
-    def __init__(self, a: float):
-        """Initialise the KernelNoTreeUnflatten class."""
-        self.a = a
-
-    def tree_flatten(self) -> tuple[tuple, dict]:
-        """
-        Flatten a pytree.
-
-        Define arrays & dynamic values (children) and auxiliary data (static values).
-        A method to flatten the pytree needs to be specified to enable JIT decoration
-        of methods inside this class.
-
-        :return: Tuple containing two elements. The first is a tuple holding the arrays
-            and dynamic values that are present in the class. The second is a dictionary
-            holding the static auxiliary data for the class, with keys being the names
-            of class attributes, and values being the values of the corresponding class
-            attributes.
-        """
-        # The score function is assumed to not change here - but it might if the kernel
-        # changes - but this does not work when kernel is specified in children
-        children = ()
-        aux_data = {"a": self.a}
-        return children, aux_data
-
-    def compute_elementwise(self, x: ArrayLike, y: ArrayLike) -> Array:
-        r"""
-        Evaluate kernel on two inputs ``x`` and ``y``.
-
-        We assume ``x`` and ``y`` are two vectors of the same dimension.
-
-        :param x: Vector :math:`\mathbf{x} \in \mathbb{R}^d`
-        :param y: Vector :math:`\mathbf{y} \in \mathbb{R}^d`
-        :return: Kernel evaluated at (``x``, ``y``)
-        """
-        return self.a * (x + y)
-
-
-class TestKernelABC(unittest.TestCase):
-    """
-    Tests related to the Kernel abstract base class in ``kernel.py``.
-    """
-
-    def setUp(self):
-        """
-        Generate data for shared use across unit tests.
-        """
-        self.default_length_scale = 1 / np.sqrt(2)
-        self.default_x = jnp.array([0.0, 1.0, 2.0, 3.0, 4.0]).reshape(-1, 1)
-        self.default_zero_kernel_row_sum = jnp.zeros(len(self.default_x))
-
-        # Define the simplest, real kernel object for testing purposes
-        self.default_kernel = coreax.kernel.SquaredExponentialKernel(
-            length_scale=self.default_length_scale
-        )
-
-    def test_approximator_valid(self) -> None:
```

Comment:
We seem to have lost this test!

Reply:
I can add this back in, but it will be deleted again if #583 is merged first as the functionality being tested will no longer exists (in this form). I had also been sceptical of the utility of this test, as it is only checking that the mocked approximator is being called, not that approximation is working (although I guess this is the only way to to have the test success decoupled from the success of the approximator implementation)